### PR TITLE
Declare core tests as CC-incompatible

### PR DESCRIPTION
**Pull request by abstratt**
This change set chooses the lesser evil: to mark tests that are known to fail under CC as skipped, as opposed to skip the entire test suite.

Note that since it is not rare for tests under CC to become flaky, marking them as expected to fail does not always help, so I defaulted to skipping, with a new skip reason. 

Issue: https://github.com/gradle/configuration-cache/issues/585

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle test RFN

**Comment by abstratt** [comment]
@bot-gradle test RFN

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/ConfigurationCacheGradleExecuter.groovy [hunk 1]
```diff
@@ -33,7 +33,7 @@ class ConfigurationCacheGradleExecuter extends DaemonGradleExecuter {
     ].collect { it.toString() }
 
     static boolean testWithLoadAfterStore() {
-        return !System.getProperty("org.gradle.configuration-cache.internal.test-disable-load-after-store")
+        return !Boolean.getBoolean("org.gradle.configuration-cache.internal.test-disable-load-after-store")
```
Comment (abstratt) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/ConfigurationCacheGradleExecuter.groovy:36]: It bothered me that `...-disable-load-after-store=false` would still disable (only "" is falsy, "false" is not).

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ToBeFixedForConfigurationCache.java [hunk 2]
```diff
@@ -81,6 +81,11 @@ enum Skip {
          * Use this reason on tests that take a long time to fail, slowing down the CI feedback.
          * Use sparingly, only in dramatic cases.
          */
-        LONG_TIMEOUT
+        LONG_TIMEOUT,
+
+        /**
+         * This test has been seen failing, but we did not have time to investigate the reason yet.
+         */
+        INVESTIGATE
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ToBeFixedForConfigurationCache.java:89]: 👍

**Review by bamboo — APPROVED** [review]
Very nice 👍

**Pull request merged**